src/toil/jobStores/fileJobStore.py

```diff
@@ -153,12 +153,19 @@ def jobs(self):
     # Functions that deal with temporary files associated with jobs
     ##########################################
 
-    def _importFile(self, otherCls, url):
+    def _importFile(self, otherCls, url, sharedFileName=None):
         if issubclass(otherCls, FileJobStore):
-            fd, absPath = self._getTempFile()
-            shutil.copyfile(self._extractPathFromUrl(url), absPath)
-            os.close(fd)
-            return self._getRelativePath(absPath)
+            if sharedFileName is None:
+                fd, absPath = self._getTempFile()
+                shutil.copyfile(self._extractPathFromUrl(url), absPath)
+                os.close(fd)
+                return self._getRelativePath(absPath)
+            else:
+                assert self._validateSharedFileName(sharedFileName)
```

Comment:
This shouldn't be an assert since passing an invalid value for sharedFileName would constitute a user error.

Change all call sites of _validateSharedFileName to conditionals whose body raises a ValueError. The exception message should contain the offending value in quotes.